resilience4j-metrics/src/main/java/io/github/resilience4j/metrics/Timer.java

```diff
@@ -126,7 +128,7 @@ default void executeRunnable(Runnable runnable) throws Exception {
      */
     static <T> CheckedFunction0<T> decorateCheckedSupplier(Timer timer, CheckedFunction0<T> supplier){
         return () -> {
-            final Context context = timer.time();
+            final Context context = timer.context();
```

Comment:
There is possibility to make `Timer` garbage free as we did it with `CircuitBreaker`.
It requires to changes contract of `Timer.onSuccess` and `Timer.onError`
We have already discussed this in Issue #117 
This is just suggestion, you can skip it if you want to.